### ISSUE TITLE
Investigate which dependencies can be made optional

### ISSUE THREAD

**Issue by lemmih** [ChainSafe/forest#2088]
**Issue summary**

Right now we have several dependencies that take a long time to build and require certain system libraries to be installed. It would be better if such dependencies were optional.

These dependencies might be optional:
- [ ] `opencl` in `fvm`. This dependency requires `ocl-icd-opencl-dev` to be installed.
- [x] `dag-pb` in `libipld`. This dependency requires `cmake` to be installed.
- [x] `snappy`, `zstd`, `zlib`, and `bzip2` in `rocksdb`. These dependencies are among the slowest to compile.

We also depend on `libcurl4`. It would be nice to avoid this and use a pure Rust alternative.

**Other information and links**

**Comment by hanabi1224** [comment]
`libcurl-sys` and underlying `libnghttp2-sys` ranks the 3rd in build times, it should be easy to replace `isahc` with sth like `reqwest`

**Comment by lemmih** [comment]
@connormullett Feel free to try your hand at one of these. If there's an unfixable reason for why, say, the fvm cannot be used without `opencl` then create a new issue marked as blocked with an explanation of what needs to happen for it to be unblocked.

**Comment by hanabi1224** [comment]
We should be able to remove `isahc`, `h2`, `curl` and `openssl` deps with https://github.com/ChainSafe/forest/pull/2140 and https://github.com/ChainSafe/forest/pull/2141

**Comment by lemmih** [comment]
@tyshko5 You can also work on this issue.

**Comment by lemmih** [comment]
I doubt we can get rid of `opencl`. Closing this issue as done.
